concent_api/middleman/middleman_server.py

```diff
@@ -137,10 +168,17 @@ def _close_middleman(self) -> None:
                 self._is_signing_service_connection_active = False
 
     async def _respond_to_user(self, data: bytes, writer: asyncio.StreamWriter) -> None:  # pylint: disable=no-self-use
-        writer.write(data)
-        await writer.drain()
+        await send_over_stream_async(data, writer)
         writer.close()
 
     def _terminate_connections(self) -> None:
         logger.info('SIGTERM received - closing connections and exiting.')
         self._loop.stop()
+
+    async def _send_immediate_error(self, writer: asyncio.StreamWriter, exception):
```

Comment:
You can also add typing for `exception: Exception`.

Reply:
Done